lazy_theta_star_planner/src/lazy_theta_star_planner.cpp

```diff
@@ -0,0 +1,153 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <vector>
+#include <memory>
+#include <string>
+#include "lazy_theta_star_planner/lazy_theta_star_planner.h"
+#include "lazy_theta_star_planner/lazy_theta_star2.h"
+
+namespace lazyThetaStarPlanner
+{
+void LazyThetaStarPlanner::configure(
+  rclcpp_lifecycle::LifecycleNode::SharedPtr parent,
+  std::string name, std::shared_ptr<tf2_ros::Buffer> tf,
+  std::shared_ptr<nav2_costmap_2d::Costmap2DROS> costmapRos)
+{
+  planner_ = std::make_unique<lazyThetaStar::LazyThetaStar>();
+  node_ = parent;
+  name_ = name;
+  planner_->costmap_ = costmapRos->getCostmap();
+  globalFrame_ = costmapRos->getGlobalFrameID();
+
+  nav2_util::declare_parameter_if_not_declared(
+    node_, name_ + ".how_many_corners", rclcpp::ParameterValue(
+      8));
+
+  node_->get_parameter(name_ + ".how_many_corners", how_many_corners_);
+
+  if (how_many_corners_ != 8) {
+    how_many_corners_ = 4;
+  }
+
+  nav2_util::declare_parameter_if_not_declared(
+    node_, name_ + ".interpolation_dist", rclcpp::ParameterValue(
+      0.1));
+
+  node_->get_parameter(name_ + ".interpolation_dist", interpolation_dist_);
+
+  nav2_util::declare_parameter_if_not_declared(
+    node_, name_ + ".lethal_cost", rclcpp::ParameterValue(
+      127));
+
+  node_->get_parameter(name_ + ".lethal_cost", lethal_cost_);
+}
+
+void LazyThetaStarPlanner::cleanup()
+{
+  RCLCPP_INFO(
+    node_->get_logger(), "CleaningUp plugin %s of type lazyThetaStarPlanner",
+    name_.c_str());
+}
+
+void LazyThetaStarPlanner::activate()
+{
+  RCLCPP_INFO(
+    node_->get_logger(), "Activating plugin %s of type lazyThetaStarPlanner",
+    name_.c_str());
+}
+
+void LazyThetaStarPlanner::deactivate()
+{
+  RCLCPP_INFO(
+    node_->get_logger(), "Deactivating plugin %s of type lazyThetaStarPlanner",
+    name_.c_str());
+}
+
+nav_msgs::msg::Path LazyThetaStarPlanner::createPlan(
+  const geometry_msgs::msg::PoseStamped & start,
+  const geometry_msgs::msg::PoseStamped & goal)
+{
+  nav_msgs::msg::Path global_path;
+  std::vector<coords<world_pts>> path;
+
+  planner_->costmap_->worldToMap(start.pose.position.x, start.pose.position.y, src_[0], src_[1]);
+  planner_->src = {static_cast<int>(src_[0]), static_cast<int>(src_[1])};
+
+  planner_->costmap_->worldToMap(goal.pose.position.x, goal.pose.position.y, dst_[0], dst_[1]);
+  planner_->dst = {static_cast<int>(dst_[0]), static_cast<int>(dst_[1])};
+
+  RCLCPP_INFO(
+    node_->get_logger(), "Got the src and dst... (%i, %i) && (%i, %i)",
+    planner_->src.x, planner_->src.y, planner_->dst.x, planner_->dst.y);
+
+  planner_->node_ = node_;
+  planner_->how_many_corners = how_many_corners_;
+
+  planner_->initializeStuff();
+  planner_->getPath(path);
```

Comment:
I think this can be semantically more clear if `getPath` returns a bool if successful so that you can do a 

```
if (planner_->getPath(path)) {... use path ...}
```

so you don't have to do the path size check and then you know if it failed to do something about it